ui/tests_ui_vfinal/test_errorchecks.js

```diff
@@ -0,0 +1,377 @@
+}
+
+/**
+ * Test: isValidDateColumn(dateColumnValue)
+ * When column specified is not one of the date columns
+ */
+function testIsValidDateColumn2() {
+  // Call the function to test
+  var generatedOutput = isValidDateColumn('Rep');
+  var expectedOutput = false;
+  
+  // Checking if generated output is same as expected output
+  assertEquals(expectedOutput, generatedOutput, 'isValidDateColumn2');
+}
+
+/**
+ * Check if the date column specified is valid
+ * (i.e is one of the table's date column)
+ * @param {string} dateColumnValue The column to be checked for validity as a date column
+ * @return {boolean} True indictes valid and false indicates invalid
+ */
+function isValidDateColumn(dateColumnValue) {
+  // Date column entered is not considered valid if the index = -1
+  var index = dateColumnNames.indexOf(dateColumnValue);
+  if(dateColumnValue === '' || index !== -1)
+    return true;
+  return false;
+}
+
+/**
+ * Test: sliceValCompatibleUtil(sliceValue,sliceOperation)
+ * When slice values and slice operation are compatible
+ */
+function testSliceValCompatible1() {
+  // Call the function to test
+  var generatedOutput = sliceValCompatibleUtil([5.85], 'Less than');
+  var expectedOutput = true;
+  
+  // Checking if generated output is same as expected output
+  assertEquals(expectedOutput, generatedOutput, 'sliceValCompatible1');
+}
+
+/**
+ * Test: sliceValCompatibleUtil(sliceValue,sliceOperation)
+ * When slice values and slice operation are not compatible
+ */
+function testSliceValCompatible2() {
+  // Call the function to test
+  var generatedOutput = sliceValCompatibleUtil(['East', 'West'], 'Equal to');
+  var expectedOutput = false;
+  
+  // Checking if generated output is same as expected output
+  assertEquals(expectedOutput, generatedOutput, 'sliceValCompatible2');
+}
+
+/**
+* Find whether the slice operation and slice values entered are compatible
+* @param {Array} sliceValue - The list of slice values entered
+* @param {string} sliceOperation - The slice operation selected by user
+* @return {boolean} True indicates compatible and false indicates incompatible
+*/
+function sliceValCompatibleUtil(sliceValue,sliceOperation) {
+  if(sliceValue.length === 1 && sliceValue[0] === '')
+    return true;
+  
+  // If slice operation is not (In,Not in),the length of sliceValue array 
+  // should not be greater than 1 
+  // For slice operations <, <=, >, >= : the slice value should be a number
+  if(sliceOperation !== 'In' && sliceOperation !== 'Not in') {
+    if(sliceValue.length > 1) {
+      return false;
+    }
+    if(sliceOperation !== 'Equal to' &&  sliceOperation !== 'Not equal to') {
+      if(isNaN(sliceValue[0])) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+/**
+ * Test: checkRangeValidity(a1Notation)
+ * When the A1Notation given is valid
+ */
+function testCheckRangeValidity1() {
+  // Call the function to test
+  var generatedOutput = checkRangeValidity('C5:20');
```

Comment:
you could actually test these with a lot of ranges, just to be 100% sure of the logic, just add a TODO() here.

Reply:
Todo stating ?

Reply:
TODO: Test valid range with more cases.